### ISSUE TITLE
[i18n] New entries contain yaml references

### ISSUE THREAD

**Issue by shumiyao** [StaticJsCMS/static-cms#937]
StaticCMS version: 3.3.9

It is not been possible to switch locale when a NEW post is created with the more recent version of StaticJsCMS.

https://github.com/StaticJsCMS/static-cms/assets/50849841/0fe69fdc-8f35-42d4-b7b0-0a850f9869e4

<details>
<summary>Click to toggle contents of `staticcms config`</summary>

```
const config: Config = {
  backend: { name: "git-gateway", branch: "main" },
  media_folder: "public/images",
  public_folder: "/images",
  i18n: {
    /**
     * Required and can be one of multiple_folders, multiple_files or single_file
     * multiple_folders - persists files in `<folder>/<locale>/<slug>.<extension>`
     * multiple_files - persists files in `<folder>/<slug>.<locale>.<extension>`
     * single_file - persists a single file in `<folder>/<slug>.<extension>`
     */
    structure: 'single_file',

    // Required - a list of locales to show in the editor UI
    locales: ['en', 'de', 'fr'],

    /**
     * Optional, defaults to the first item in locales.
     * The locale to be used for fields validation and as a baseline for the entry.
     */
    defaultLocale: 'en'
  },
  collections: [
 {
      name: "i18n-posts",
      label: "i18n Posts",
      folder: "content/i18n-posts/",
      extension: "mdx",
      format: "frontmatter",
      create: true,
      slug: "{{slug}}",
      identifier_field: "slug",
      summary: "{{title}}",
      i18n: true,
      fields: [
        { label: "Slug", name: "slug", widget: "string" },
        { label: "Title", name: "title", widget: "string", i18n: true },
        {
          label: "Publish Date",
          name: "date",
          widget: "datetime",
          format: "yyyy-MM-dd",
          date_format: "yyyy-MM-dd",
          time_format: false,
        },
        {
          label: "Author",
          name: "author",
          widget: "relation",
          collection: "meta",
          file: "authors",
          search_fields: ["authors.*.name"],
          display_fields: ["authors.*.name"],
          value_field: "authors.*.slug",
        },
        {
          label: "Tags",
          label_singular: "Tag",
          name: "tags",
          widget: "list",
          summary: "{{fields.tag}}",
          fields: [
            {
              label: "Tag",
              name: "tag",
              widget: "relation",
              collection: "meta",
              file: "tags",
              search_fields: ["tags.*.name"],
              display_fields: ["tags.*.name"],
              value_field: "tags.*.slug",
            },
          ],
        },
        { label: "Body", name: "body", widget: "markdown", i18n: true },
      ],
    },
  ],
};
```
</details>

<details>
<summary>Saved MDX file content</summary>
---
en:
  date: 2023-10-13
  slug: test
  title: TEST EN
  body: |
    TEsT EN
  author: kanefreeman
  tags:
    - document
de: &a1
  date: 2023-10-13
  body: |
    TEST DE
fr: *a1
---
</details>

<details>
<summary>The locale change works with this MDX data saved with the staticcms version 3.3.3</summary>
---
en:
  date: 2023-09-24
  slug: test-post
  title: Test Post EN
  author: kanefreeman
  tags:
    - document
  body: |
    Test Post EN
de:
  date: 2023-09-24
  body: |
    Test Post DE
  title: Test Post DE
fr:
  body: |
    test french
  title: Test Post FR
---
</details>

**Comment by KaneFreeman** [comment]
The format on your `test.mdx` (the post titled `TEST EN`) has a reference in it between the German and French translations. That would explain why it to not be switching locales. It is actually duplicating the data when its read in so both translations are the same. The app is correctly switching locales.

```yaml
en:
  date: 2023-10-13
  slug: test
  title: TEST EN
  body: |
    TEsT EN
  author: kanefreeman
  tags:
    - document
de: &a1
  date: 2023-10-13
  body: |
    TEST DE
fr: *a1
```

**Comment by shumiyao** [comment]
Sorry but how can we avoid the "duplicate" mode when creating a new post collection? It seems to creating new files in the duplicate mode in the recent versions.

**Comment by KaneFreeman** [comment]
Ah yeah you are right .... wonder why its doing that.

**Comment by KaneFreeman** [comment]
This is fixed in `v3.3.10`

**Comment by KaneFreeman** [comment]
There appears an additional issue here. I'm reopening this.

**Comment by KaneFreeman** [comment]
NOW, this is resolved in `v3.4.0`
